src/type-generator.ts

```diff
@@ -528,21 +529,27 @@ export class TypeGenerator {
           throw new Error('only enums with string or number values are supported');
         }
 
-        // sluggify and turn to UPPER_SNAKE_CASE
-        let memberName = snakeCase(`${value}`.replace(/[^a-z0-9]/gi, '_')).split('_').filter(x => x).join('_').toUpperCase();
+        let memberName;
 
-        // If enums of same value exists, then we choose one of them and skip adding others
-        // since that would cause conflict
-        const lowerCaseValue = value?.toString().toLowerCase();
-        if (lowerCaseValue && !processedValues.has(lowerCaseValue)) {
-          processedValues.add(lowerCaseValue);
+        if (value && isAllowlistedCharacter(value.toString())) {
```

Comment:
I like the idea, taking a look. 

For context, we import Custom Resource Definitions(these are json schemas) in cdk8s and there is an issue with importing an enum in Prometheus CRD: https://github.com/prometheus-operator/kube-prometheus/blob/373e2b415bd0ddbcb049dea54db80c2a78b76945/manifests/setup/0alertmanagerConfigCustomResourceDefinition.yaml#L67-L71

It does not have these symbols occurring between other words but it makes sense to add this. My thought process here was to just add an allowlist so that we can add to that list if we agree with providing an exception for the user.